src/printer.js

```diff
@@ -3880,7 +3880,23 @@ function makeString(rawContent, enclosingQuote) {
     return match;
   });
 
-  return enclosingQuote + newContent + enclosingQuote;
+  // Matches any unnecessarily escaped character.
+  // Adapted from https://github.com/eslint/eslint/blob/de0b4ad7bd820ade41b1f606008bea68683dc11a/lib/rules/no-useless-escape.js#L27
+  const regexUnnecessaryStringEscapes = /((?:^|[^\\])(?:\\\\)*)\\([^\\nrvtbfux\r\n\u2028\u2029"'0-7])/g;
+
+  let minimallyEscapedContent = newContent;
+  let minimallyEscapedContentPrev;
+
+  while (minimallyEscapedContent !== minimallyEscapedContentPrev) {
```

Comment:
Could you add a max iteration length as a safe guard. I don't want it to go into an infinite loop if there's an edge case that is not well supported.